tests/test_items.py

```diff
@@ -65,4 +90,105 @@ class Number(Item):
     assert actual == expected
 
     assert Number.from_list(None) == []
-    assert Number.from_list([]) == []
\ No newline at end of file
+    assert Number.from_list([]) == []
+
+
+def assert_all_isinstance(lst, cls):
+    assert all(isinstance(el, cls) for el in lst)
+
+
+def assert_pagination_types(item):
+    assert isinstance(item.paginationNext, PaginationLink)
+    assert isinstance(item.paginationPrevious, PaginationLink)
+
+
+# XXX: should we make tests below to pick up types based on type annotations,
```

Comment:
>  should we make tests below to pick up types based on type annotations instead of hardcoding all attributes?

@kmike I've implemented automatic type checking here https://github.com/scrapinghub/autoextract-poet/pull/14/commits/30d499e77b3f86a8835d22c8fc7e3dcde67e860d but I wouldn't remove the tests you introduced because they are more strict (detect the absence of data when expected) and redundancy is positive in unit tests.